akkaConstruct/src/test/scala/akka/contrib/construct/ActorOfSpec.scala

```diff
@@ -119,6 +112,22 @@ object ActorOfSpec {
         child.tell(GetChildPath, forward)
     }
   }
+  
+  object CustomChildMaker {
+    class Construct(override val defaultArgs: Seq[String], useDefault: Boolean) extends OneParamConstructActor with DefaultConstructArgs {
+      override type ActorClass = CustomChildMaker
+      override type P1 = Seq[String]
+      override def actorOf(args: Seq[String])(implicit tag: ClassTag[CustomChildMaker]): ActorOf[CustomChildMaker] = {
+        ActorOf(construct(args))
+      }
+      override protected val construct: (Seq[String]) => CustomChildMaker = {
+        // Intercept construction logic
+        if (useDefault) (_: Args) => new CustomChildMaker(defaultArgs, Child.Construct)
+        else (path: Seq[String]) => new CustomChildMaker(path, Child.Construct)
+      }
+    }
+    object Construct extends Construct(Seq("default"), useDefault = true)
```

Comment:
This is the new pattern. Instead of using `ActorOf[T]`, you make a `class ConstructT { type ActorClass = T }` and you can add some `ParamsConstructActor` or just a parameterless `ConstructActor`. If you add a `DefaultConstructArgs`, then you have to provide the default, but it allows you to call `.actor.ref` with no arguments. Additionally, you can still request for an `ActorOf[T]` if you don't plan to modify its construction.